src/status_im2/contexts/quo_preview/wallet/transaction_progress.cljs

```diff
@@ -0,0 +1,155 @@
+   (preview/customization-color-option)])
+
+(def total-box 85)
+(def counter (reagent/atom 0))
+(def interval-id (reagent/atom nil))
+(def interval-ms 50)
+
+(defn- stop-interval
+  []
+  (when @interval-id
+    (js/clearInterval @interval-id)
+    (reset! interval-id nil)))
+
+(defn- clear-counter
+  []
+  (reset! counter 0))
+
+(defn- update-counter
+  [state]
+  (let [new-counter-value (inc @counter)]
+    (if (or (and (= state :pending) (> new-counter-value 0))
+            (and (= state :sending) (> new-counter-value 2))
+            (and (= state :confirmed) (> new-counter-value 4))
+            (and (= state :finalising) (> new-counter-value 18))
+            (and (= state :finalized) (> new-counter-value total-box))
+            (and (= state :error) (> new-counter-value 2)))
+      (stop-interval)
+      (reset! counter new-counter-value))))
+
+(defn- start-interval
+  [state]
+  (reset! interval-id
+    (js/setInterval
+     (fn []
+       (update-counter state))
+     interval-ms)))
+
+(defn- get-networks
+  [state]
+  (case (:network state)
+    :mainnet           [{:network      :mainnet
+                         :state        (:state-mainnet state)
+                         :counter      counter
+                         :total-box    total-box
+                         :epoch-number (:epoch-number-mainnet state)}]
+    :optimism          [{:network      :optimism
+                         :state        (:state-optimism state)
+                         :progress     (:optimism-progress-percentage state)
+                         :epoch-number (:epoch-number-optimism state)}]
+    :arbitrum          [{:network      :arbitrum
+                         :state        (:state-arbitrum state)
+                         :progress     (:arbitrum-progress-percentage state)
+                         :epoch-number (:epoch-number-arbitrum state)}]
+    :optimism-arbitrum [{:network      :optimism
+                         :state        (:state-optimism state)
+                         :progress     (:optimism-progress-percentage state)
+                         :epoch-number (:epoch-number-optimism state)}
+                        {:network      :arbitrum
+                         :state        (:state-arbitrum state)
+                         :progress     (:arbitrum-progress-percentage state)
+                         :epoch-number (:epoch-number-arbitrum state)}]))
+
+(defn- f-view
+  [state]
+  (fn []
+    (rn/use-effect
+     (fn []
+       (start-interval (:state-mainnet @state))
+       (clear-counter)
+       (fn []
+         (stop-interval)))
+     [(:state-mainnet @state)])
+    [preview/preview-container {:state state :descriptor descriptor}
+     [quo/transaction-progress
+      (assoc @state
+             :networks
+             (get-networks @state))]]))
+
+(defn view
+  []
+  (let [state (reagent/atom
+               {:title                        "Title"
+                :counter                      counter
```

Comment:
I can do sth. Just pass a number to the component as progress-value, and then if the animations is needed, handle it in the preview as it is right now. @J-Son89 Let me know what you think

Reply:
Yep that also works, mostly just that the component itself should not need to know if the prop is a ratom or not because that makes it slightly more confusing

Reply:
Done